python/pyiceberg/exceptions.py

```diff
@@ -90,3 +90,7 @@ class SignError(Exception):
 
 class ResolveError(Exception):
     pass
+
+
+class StaticTableImmutableError(Exception):
```

Comment:
How do you feel like re-using the internal `NotImplementedError`?

Reply:
My reasoning was that for a `StaticTable` refreshing metadata doesn't make sense, hence it's never going to be implemented, hence `NotImplementedError` doesn't make sense. 

I might have misunderstood the whole purpose of `refresh()`.